packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -679,11 +706,12 @@ export default function(
     }
 
     // In order to support react-lifecycles-compat polyfilled components,
-    // Unsafe lifecycles should not be invoked for any component with the new gDSFP.
+    // Unsafe lifecycles should not be invoked for components using the new APIs.
     if (
+      typeof ctor.getDerivedStateFromProps !== 'function' &&
+      typeof instance.getSnapshotBeforeUpdate !== 'function' &&
```

Comment:
We seem to repeat these checks quite a few times. Can we extract them to cached variables, and use them throughout the method?

Reply:
Sure, I removed the redundant checks from the two methods that had them (88d4cac)